tweetflask.py

```diff
@@ -29,62 +29,49 @@ def get():
     # looking up the username in the DB
     for user in jsondata['UserData']:
         if username.upper().lower() == user['Username'].upper().lower():
-            return jsonify(get_tweet(jsondata, user, mood, None))
+            return jsonify(get_tweet(jsondata, user, mood))
     # if the username does not exist in DB this line will be executed
     # to return an empty array
-    return jsonify(get_tweet(jsondata, None, None, None))
+    return jsonify(get_tweet(jsondata, None, None))
 
 
-@app.route('/tweets/<get_id>')
-def get_ID(get_id):
-    # Opening and reading the db file
-    jsondata = read_file()
-
-    for tweets in jsondata['TweetData']:
-        if get_id == tweets['TweetID']:
-            for user in jsondata['UserData']:
-                if tweets['UserID'] == user['UserID']:
-                    return jsonify(get_tweet(jsondata, user, None, get_id))
-    return jsonify(errors('404', 'tweet does not exist')), 404
+@app.route('/tweets/<tweet_id>')
+def get_ID(tweet_id):
+    try:
+        return jsonify(tweets[tweet_id])
+    except KeyError:
+        return jsonify(errors('404', 'tweet does not exist')), 404
 
 
 # looks up all the tweets that belong to the username provided
 # and looks for the mood if provided
-def get_tweet(jsondata, user, mood, get_id):
+def get_tweet(jsondata, user, mood):
     tweet_data = {
         'data': [
         ]
     }
-
     if user is None:
         return tweet_data
     for tweets in jsondata['TweetData']:
```

Comment:
You are declaring a local variable `tweets` here which is as same as the variable name of your tweets dictionary. It may cause potential conflict issues.